packages/lodestar/src/chain/blocks/processor.ts

```diff
@@ -58,14 +61,28 @@ export class BlockProcessor {
  */
 export async function processBlockJob(modules: BlockProcessorModules, job: IBlockJob): Promise<void> {
   try {
-    // First convert incoming blocks to TreeBacked backing (for efficiency reasons) if it's a regular block
-    // The root is computed multiple times, the contents are hash-tree-rooted multiple times,
-    // and some of the contents end up in the state as tree-form.
-    if (!isTreeBacked(job.signedBlock)) {
-      job.signedBlock = modules.config.types.SignedBeaconBlock.tree.createValue(job.signedBlock);
-    }
-    await validateBlock({...modules, job});
-    await processBlock({...modules, job});
+    await validateBlocks({...modules, jobs: [job]});
+    await processBlocks({...modules, jobs: [job]});
+  } catch (e) {
+    // above functions only throw BlockError
+    modules.emitter.emit(ChainEvent.errorBlock, e);
+  }
+}
+
+/**
+ * Similar to processBlockJob but this process a chain segment
+ */
+export async function processChainSegmentJob(
+  modules: BlockProcessorModules,
+  chainSegmentJob: IChainSegmentJob
+): Promise<void> {
+  try {
+    const blockJobs: IBlockJob[] = chainSegmentJob.signedBlocks.map((signedBlock) => ({
+      signedBlock,
+      ...chainSegmentJob,
+    }));
+    await validateBlocks({...modules, jobs: blockJobs});
+    await processBlocks({...modules, jobs: blockJobs});
   } catch (e) {
     // above functions only throw BlockError
     modules.emitter.emit(ChainEvent.errorBlock, e);
```

Comment:
I think this should throw an error here, otherwise consumers of `processChainSegment` won't know if it was successful or not